lib/execution_engine2/utils/SDKMethodRunner.py

```diff
@@ -261,3 +264,45 @@ def run_job(self, params, ctx):
         logging.info(condor_job_id)
         logging.info(type(condor_job_id))
         return condor_job_id
+
+    def get_job_params(self, job_id):
+        job_params = dict()
+        job_config = dict()
+
+        with self.get_mongo_util().me_collection(self.config["mongo-jobs-collection"]):
+
+            try:
+                job = Job.objects(id=job_id)[0]
+            except Exception:
+                raise ValueError("Unable to find job:\nError:\n{}".format(traceback.format_exc()))
+
+            job_input = job.job_input
+
+            job_params["method"] = job_input.method
+            job_params["params"] = job_input.params
+            job_params["service_ver"] = job_input.service_ver
+            job_params["app_id"] = job_input.app_id
+            job_params["wsid"] = job_input.wsid
+            job_params["parent_job_id"] = job_input.parent_job_id
+            job_params["source_ws_objects"] = job_input.source_ws_objects
+
+        return job_params, job_config
+
+    def update_job_status(self, job_id, status):
+
+        if not (job_id and status):
+            raise ValueError("Please provide both job_id and status")
+
+        with self.get_mongo_util().me_collection(self.config["mongo-jobs-collection"]):
+
+            try:
+                job = Job.objects(id=job_id)[0]
+            except Exception:
+                raise ValueError("Unable to find job:\nError:\n{}".format(traceback.format_exc()))
+
+            job.status = status
+            job.updated = datetime.utcnow()
```

Comment:
Maybe we should edit the model so we don't have to do job.updated manually each time

https://stackoverflow.com/questions/12230022/update-datetime-field-in-mongoengine-with-auto-now-add

Reply:
👍 sounds like a good idea. Implemented auto update upon saving.